lib/cloud_controller/blobstore/storage_cli/storage_cli_client.rb

```diff
@@ -16,29 +16,108 @@ class << self
         attr_reader :registry
 
         def register(provider, klass)
-          registry[provider] = klass
+          registry[provider.to_s] = klass
         end
 
-        def build(connection_config:, directory_key:, root_dir:, min_size: nil, max_size: nil)
-          provider = connection_config[:provider]
-          raise 'Missing connection_config[:provider]' if provider.nil?
+        def build(directory_key:, root_dir:, resource_type: nil, min_size: nil, max_size: nil)
+          raise 'Missing resource_type' if resource_type.nil?
 
-          impl_class = registry[provider]
+          cfg = fetch_and_validate_config!(resource_type)
+          provider = cfg['provider']
+
+          key        = provider.to_s
+          impl_class = registry[key] || registry[key.downcase] || registry[key.upcase]
           raise "No storage CLI client registered for provider #{provider}" unless impl_class
 
-          impl_class.new(connection_config:, directory_key:, root_dir:, min_size:, max_size:)
+          impl_class.new(provider:, directory_key:, root_dir:, resource_type:, min_size:, max_size:)
+        end
+
+        def fetch_and_validate_config!(resource_type)
+          path = config_path_for!(resource_type)
+
+          begin
+            json = Oj.load(File.read(path))
+          rescue StandardError => e
+            raise BlobstoreError.new("Failed to parse storage-cli config JSON at #{path}: #{e.message}")
+          end
+
+          validate_required_keys!(json, path)
+          json
+        end
+
+        def config_path_for!(resource_type)
+          key =
+            case resource_type.to_s
+            when 'droplets', 'buildpack_cache' then :storage_cli_config_file_droplets
+            when 'buildpacks'                  then :storage_cli_config_file_buildpacks
+            when 'packages'                    then :storage_cli_config_file_packages
+            when 'resource_pool'               then :storage_cli_config_file_resource_pool
+            else
+              raise BlobstoreError.new("Unknown resource_type: #{resource_type}")
+            end
+
+          path = VCAP::CloudController::Config.config.get(key)
```

Comment:
What if `config.get(key)` returns nil? Maybe include `key` in the error message?